### PR TITLE
Add Stillbirth as a pre-defined event type

### PR DESCRIPTION
**Pull request by DaveSch-gramps**
Add Stillborn as a fallback event for both Birth and Death

### CODE REVIEW

**Comment by DaveSch-gramps** [comment]
Feature Request #0011554

**Comment by Nick-Hall** [comment]
The new Gedcom 7.0 standard suggests using the TYPE tag to provide further classification of an event.  “Stillborn” as a qualifier to a BIRT event is given as a specific example:

```
1 BIRT
2 TYPE Stillborn
```

**Comment by Nick-Hall** [comment]
In the associated Discourse [topic](https://gramps.discourse.group/t/stillborn-as-a-coded-event-type/181), four people were in favour of this change.  One stated that he would continue to use two separate events, but I don't see that he would be affected by this change.

File: gramps/gen/lib/eventtype.py [hunk 1]
```diff
@@ -253,7 +256,8 @@ def _T_(value, context=''): # enable deferred translations
         ENGAGEMENT: _T_("engd.", "Engagement abbreviation"),
         DIVORCE: _T_("div.", "Divorce abbreviation"),
         DIV_FILING: _T_("div.f.", "Divorce Filing abbreviation"),
-        ANNULMENT: _T_("annul.", "Annulment abbreviation")
+        ANNULMENT: _T_("annul.", "Annulment abbreviation"),
+        STILLBIRTH: _T_("Stillbirth abbreviation|still.")
```
Comment (Nick-Hall) [inline gramps/gen/lib/eventtype.py:260]: Needs to use the new translation context:

`STILLBIRTH: _T_("still.", "Stillbirth abbreviation")`

File: gramps/gen/lib/eventtype.py [hunk 2]
```diff
@@ -280,6 +284,13 @@ def is_death(self):
         """
         return self.value == self.DEATH
 
+    def is_stillborn(self):
```
Comment (Nick-Hall) [inline gramps/gen/lib/eventtype.py:287]: Is this method needed?

File: gramps/gui/views/treemodels/peoplemodel.py [hunk 3]
```diff
@@ -265,7 +265,9 @@ def _get_birth_data(self, data, sort_mode):
             event = self.db.get_event_from_handle(er.ref)
             etype = event.get_type()
             date_str = get_date(event)
-            if (etype in [EventType.BAPTISM, EventType.CHRISTEN]
+            if (etype in [EventType.STILLBIRTH,
```
Comment (Nick-Hall) [inline gramps/gui/views/treemodels/peoplemodel.py:268]: Use `etype.is_birth_fallback()` instead.

File: gramps/gui/views/treemodels/peoplemodel.py [hunk 4]
```diff
@@ -322,7 +324,8 @@ def _get_death_data(self, data, sort_mode):
             event = self.db.get_event_from_handle(er.ref)
             etype = event.get_type()
             date_str = get_date(event)
-            if (etype in [EventType.BURIAL,
+            if (etype in [EventType.STILLBIRTH,
```
Comment (Nick-Hall) [inline gramps/gui/views/treemodels/peoplemodel.py:327]: Use `etype.is_death_fallback()` instead.

File: gramps/gen/lib/eventtype.py (hunk 1 above)
Comment (DaveSch-gramps) [reply gramps/gen/lib/eventtype.py]: Fixed in [15a0085](https://github.com/gramps-project/gramps/pull/1148/commits/15a0085b6316bb277400a9133eabf32b47be2558)

File: gramps/gen/lib/eventtype.py (hunk 2 above)
Comment (DaveSch-gramps) [reply gramps/gen/lib/eventtype.py]: Removed the def section and it still worked so obviously was not needed.

Fixed in [15a0085](https://github.com/gramps-project/gramps/pull/1148/commits/15a0085b6316bb277400a9133eabf32b47be2558)

File: gramps/gui/views/treemodels/peoplemodel.py (hunk 3 above)
Comment (DaveSch-gramps) [reply gramps/gui/views/treemodels/peoplemodel.py]: The same issue appeared at line 367 so also made that change.

Fixed in [15a0085](https://github.com/gramps-project/gramps/pull/1148/commits/15a0085b6316bb277400a9133eabf32b47be2558)

File: gramps/gui/views/treemodels/peoplemodel.py (hunk 4 above)
Comment (DaveSch-gramps) [reply gramps/gui/views/treemodels/peoplemodel.py]: The same issue appeared at line 407

Fixed in [15a0085](https://github.com/gramps-project/gramps/pull/1148/commits/15a0085b6316bb277400a9133eabf32b47be2558)

**Comment by Nick-Hall** [comment]
I asked for feedback on the [mailing list](https://sourceforge.net/p/gramps/mailman/message/37631851/).  The only reply pointed out that there are some places where fallback events are not used.  We may need to check this, but I don't see any reason not to merge this PR now.

**Pull request merged**